File: struct-type/10-composite/java/Main.java
package main;

public class Main {
    
    public static void main(String[] args) {
       
        // 创建一个简单的 DOM 树
        ElementNode html = new ElementNode("html");
        
        ElementNode body = new ElementNode("body");
        
        ElementNode div = new ElementNode("div");
        div.setAttribute("class", "container");
        
        ElementNode p = new ElementNode("p");
        p.appendChild(new TextNode("Hello, World!"));
        
        div.appendChild(p);
        body.appendChild(div);
        html.appendChild(body);
        
        // 渲染 DOM 树
        html.render();
    }
}

File: struct-type/10-composite/go/main.go
package main

func main() {

	// 创建 DOM 树
	div := NewElementNode("div")
	div.SetAttribute("class", "container")

	p1 := NewElementNode("p")
	p1.AppendChild(NewTextNode("这是第一个段落"))

	p2 := NewElementNode("p")
	p2.SetAttribute("style", "color: blue")
	p2.AppendChild(NewTextNode("这是第二个段落"))

	div.AppendChild(p1)
	div.AppendChild(p2)

	// 渲染整个 DOM 树
	div.Render()
}
